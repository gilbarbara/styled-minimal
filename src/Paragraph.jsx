import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { isDefined, spacer } from './utils/helpers';

import Box from './Box';

const Paragraph = styled(Box)(
  ({ mb, mt }) => css`
    margin-bottom: ${isDefined(mb) ? mb : 0};
    margin-top: ${isDefined(mt) ? mt : 0};

    & + & {
      margin-top: ${spacer(2)};
    }
  `,
);

Paragraph.displayName = 'Paragraph';

Paragraph.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  ...Box.basePropTypes,
};

Paragraph.defaultProps = {
  as: 'p',
};

export default Paragraph;
