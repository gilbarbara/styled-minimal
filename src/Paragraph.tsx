import * as React from 'react';
import styled, { css } from 'styled-components';

import { spacer } from './utils/helpers';
import { baseStyles, sx, system } from './utils/system';

import { ParagraphProps } from './types';

export const StyledParagraph = styled('p')<ParagraphProps>(
  {
    ...baseStyles,
    lineHeight: 1.4,
  },
  () => css`
    margin-bottom: 0;
    margin-top: 0;

    & + & {
      margin-top: ${spacer(2)};
    }
  `,
  sx,
  system,
);

export const Paragraph = React.forwardRef<HTMLParagraphElement, ParagraphProps>((props, ref) => (
  <StyledParagraph ref={ref} {...props} />
));
