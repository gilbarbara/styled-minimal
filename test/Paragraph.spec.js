import React from 'react';
import { mount } from 'enzyme';

import Paragraph from '../src/Paragraph';

describe('Paragraph', () => {
  const wrapper = mount(
    <React.Fragment>
      <Paragraph>
        Far far away, behind the word mountains, far from the countries Vokalia and Consonantia,
        there live the blind texts.
      </Paragraph>
      <Paragraph>
        Separated they live in Bookmarksgrove right at the coast of the Semantics, a large language
        ocean. A small river named Duden flows by their place and supplies it with the necessary
        regelialia. It is a paradisematic country, in which roasted parts of sentences fly into your
        mouth.
      </Paragraph>
    </React.Fragment>,
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
