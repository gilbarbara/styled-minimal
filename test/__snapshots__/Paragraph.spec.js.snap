// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph should render properly 1`] = `
Array [
  .c1 {
  box-sizing: border-box;
  margin: 0;
}

.c1 + .c0 {
  margin-top: 8px;
}

<Paragraph
    as="p"
  >
    <p
      className="c0 c1"
    >
      Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts.
    </p>
  </Paragraph>,
  .c1 {
  box-sizing: border-box;
  margin: 0;
}

.c1 + .c0 {
  margin-top: 8px;
}

<Paragraph
    as="p"
  >
    <p
      className="c0 c1"
    >
      Separated they live in Bookmarksgrove right at the coast of the Semantics, a large language ocean. A small river named Duden flows by their place and supplies it with the necessary regelialia. It is a paradisematic country, in which roasted parts of sentences fly into your mouth.
    </p>
  </Paragraph>,
]
`;
